proxy/http/HttpSM.cc

```diff
@@ -2439,8 +2439,12 @@ HttpSM::state_cache_open_write(int event, void *data)
   if (ua_session) {
     NetVConnection *vc = ua_session->get_netvc();
     if (vc && vc->thread != this_ethread()) {
-      pending_action = vc->thread->schedule_imm(this, event, data); // Stay on the same thread!
-      return 0;
+      if (vc->thread->is_event_type(ET_NET) || vc->thread->is_event_type(SSLNetProcessor::ET_SSL)) {
+        pending_action = vc->thread->schedule_imm(this, EVENT_INTERVAL);
```

Comment:
I can confirm I get a segfault in some cases from this. Changing line 2443 to `pending_action = vc->thread->schedule_imm(this, event);` fixes that.